artifacts.go

```diff
@@ -418,12 +381,48 @@ func resolve(base *BaseArtifact, artifactType string, path string, contentType s
 			contentType = "application/octet-stream"
 		}
 	}
+	// Is content encoding specified in task payload?
+	if contentEncoding == "" {
+		extension := filepath.Ext(path)
+		// based on https://github.com/evansd/whitenoise/blob/03f6ea846394e01cbfe0c730141b81eb8dd6e88a/whitenoise/compress.py#L21-L29
+		// with .7z added (useful for NSS)
+		SkipCompressionExtensions := map[string]bool{
+			// Images
+			".jpg":  true,
+			".jpeg": true,
+			".png":  true,
+			".gif":  true,
+			".webp": true,
+			// Compressed files
+			".7z":  true,
+			".zip": true,
+			".gz":  true,
+			".tgz": true,
+			".bz2": true,
+			".tbz": true,
+			".whl": true, // Python wheel are already zip file
+			".xz":  true,
+			// Flash
+			".swf": true,
+			".flv": true,
+			// Fonts
+			".woff":  true,
+			".woff2": true,
+		}
+		if SkipCompressionExtensions[extension] {
+			// fallback to encoding specified by file extension
+			contentEncoding = strings.Trim(extension, ".")
+		} else {
+			// lastly, fallback to standard gzip
+			contentEncoding = "gzip"
+		}
```

Comment:
Lines 412-418 should be:

```
		// When the task payload does not explicitly set contentEncoding,
		// "identity" should be used when the file extension is blacklisted
		// in SkipCompressionExtensions, otherwise "gzip".
		if SkipCompressionExtensions[extension] {
			contentEncoding = "identity"
		else {
			contentEncoding = "gzip"
		}
```